### DISCUSSION TITLE
How to add authentication to generated CRUD endpoints?

### DISCUSSION THREAD

**Discussion by gtamas** [loopbackio/loopback-next#7234] [Q&A]
Hi

I generated stuff with lb4 rest-crud. My app also uses @loopback/authentication-jwt

How can I add @authenticate("jwt") decorator to the generated endpoints?

Thanks

Regards

Tamas

**Comment by dougal83** [comment]
Hey, does the following answer your question? (see step 3b/4) 

https://loopback.io/doc/en/lb4/Authentication-tutorial.html#3b-add-endpoints-in-usercontroller

**Reply by gtamas** [reply]
Hi 

Thanks for the info, but this is not what i need.

The generated endpoints don't really exist as code, so I can't put the decorator there. 

When I generate rest-crud, all I get is a config file which looks like this:

model-endpoints/user.res-config.ts

```
const config: ModelCrudRestApiConfig = {
  model: User,
  pattern: 'CrudRest',
  dataSource: 'mongo',
  basePath: '/useradmin',
};
```

How do I customise this?

**Reply by dougal83** [reply]
Hey, sorry I haven't been around for a while and the rest-crud feature is new to me.  How would defining a default authentication strategy like the following work for you?

https://loopback.io/doc/en/lb4/Authentication-component-options.html

**Reply by gtamas** [reply]
Hi

OK thanks I will check this.
